backend/internal/pkg/repositories/parkingspot/postgres.go

```diff
@@ -82,6 +83,129 @@ func (p *PostgresRepository) Create(ctx context.Context, userID int64, spot *mod
 	return entry, availability, nil
 }
 
+func (p *PostgresRepository) UpdateByUUID(ctx context.Context, spotID uuid.UUID, updateSpot *models.ParkingSpotUpdateInput) (Entry, []models.TimeUnit, error) {
+	tx, err := p.db.BeginTx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return Entry{}, nil, fmt.Errorf("could not start a transaction: %w", err)
+	}
+	defer func() { _ = tx.Rollback() }() // Default to rollback if commit is not done
+
+	spotSetter, err := spotSetterFromUpdateInput(updateSpot)
+	if err != nil {
+		return Entry{}, nil, err
+	}
+
+	// Update the spot details
+	updated, err := dbmodels.Parkingspots.UpdateQ(
+		ctx, p.db,
+		dbmodels.UpdateWhere.Parkingspots.Parkingspotuuid.EQ(spotID),
+		spotSetter,
+		um.Returning(dbmodels.Parkingspots.Columns()),
+	).One()
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return Entry{}, []models.TimeUnit{}, ErrNotFound
+		}
+		return Entry{}, []models.TimeUnit{}, fmt.Errorf("could not execute update: %w", err)
+	}
+
+	// Delete unbooked times
+	_, err = dbmodels.Timeunits.DeleteQ(
+		ctx, p.db,
+		dbmodels.DeleteWhere.Timeunits.Parkingspotid.EQ(updated.Parkingspotid),
+		dbmodels.DeleteWhere.Timeunits.Bookingid.IsNull(),
+	).Exec()
```

Comment:
Thanks for the suggestion, however the issue is that in order for us to know if we are deleting a booked time, we would have to query the existing time units with a non-NULL booking id and look for its absense in the availability input since we can't rely on a constraint error here to let us know.  I'm not sure this would help us save any cycles and make the operation less blocking.  Please let me know if I am misunderstanding.